### ISSUE TITLE
installation with macport

### ISSUE THREAD

**Issue by fredRos** [fredRos/pypmc#44]
On mac os 10.10.3 with macports and python 2.7.9

Installation from tarball does not work

```
$ pip install pypmc
Collecting pypmc
  Downloading pypmc-1.0.tar.gz (847kB)
    100% |████████████████████████████████| 847kB 513kB/s 
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 20, in <module>
      File "/private/var/folders/bz/4y0xq0m10gxfvg_4w7s947lw0000gn/T/pip-build-IWz58P/pypmc/setup.py", line 41, in <module>
        ext_modules = cythonize(extensions, compiler_directives=compiler_directives)
      File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/Cython/Build/Dependencies.py", line 754, in cythonize
        aliases=aliases)
      File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/Cython/Build/Dependencies.py", line 649, in create_extension_list
        for file in nonempty(extended_iglob(filepattern), "'%s' doesn't match any files" % filepattern):
      File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/Cython/Build/Dependencies.py", line 103, in nonempty
        raise ValueError(error_msg)
    ValueError: 'pypmc/*/*.pyx' doesn't match any files
    
    ----------------------------------------
    Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/bz/4y0xq0m10gxfvg_4w7s947lw0000gn/T/pip-build-IWz58P/pypmc
```

installation from git clone works but import pypmc does not

```
$ python -c "import pypmc"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "pypmc/__init__.py", line 9, in <module>
    from . import mix_adapt, density, sampler, tools
  File "pypmc/mix_adapt/__init__.py", line 7, in <module>
    from . import hierarchical, variational, pmc, r_value
  File "pypmc/mix_adapt/r_value.py", line 7, in <module>
    from ..tools._doc import _add_to_docstring
  File "pypmc/tools/__init__.py", line 6, in <module>
    from ._partition import partition, patch_data
  File "pypmc/tools/_partition.py", line 6, in <module>
    from ..density.gauss import Gauss
  File "pypmc/density/__init__.py", line 6, in <module>
    from . import gauss, student_t, mixture, base
ImportError: cannot import name gauss
```

**Comment by danielrosenbloom** [comment]
I got the same ValueError when installing from the tarball, as there are indeed no *.pyx files in the required folders! When I install from the latest on git, I have to do "sudo python setup.py install", but importing pypmc indeed works for me.

Without sudo, I get the error "~/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/easy-install.pth: Permission denied." With sudo, the install script successfully adds pypmc and the required versions of setuptools, scipy, and numpy to the easy-install.pth.

Using OSX 10.10.3 and Enthought Canopy Python 2.7.6.

**Comment by jPhy** [comment]
There are no "*.pyx" files in the distribution since they are only necessary for developers who should clone the git-repo.

The problem is that the setup assumes if "pypmc/*/*.pyx" matches no file then an empty list of extensions is returned without error. (That is indeed the case on ubuntu14.04/13.10 using bash.) When no "*.pyx" is found, setup should proceed using the cythonized "*.c" files that are available in the source-distribution (taball).
This should be fixed in 0c66c53e16d273579697cffda5b73d6c1d98e81e but I cannot check since that error never occurred on my system.

@danielrosenbloom
Did you try "python setup.py install --user"? If you omit both, the "--user" and the "sudo", python tries to install into system-directories where ordinary users often have no write permission.

**Comment by fredRos** [comment]
I can't test this on my linux machine but it should be fixed now with 47e9f0f
